roles/upload-logs-otc-swift/library/zuul_swift_upload.py

```diff
@@ -782,6 +800,10 @@ def _post_file(self, file_detail):
             }
         )
 
+        # ensure file was uploaded
+        if not self._file_exist(relative_path, file_detail):
+            raise Exception("Swift upload failed")
```

Comment:
uhm, do we really "fail" the job itself if one of the artifacts was not uploaded? Maybe retry, log somethings, but not simply abort

Reply:
`_post_file` is wrapped in `retry_function`, retrying on any `Exception`. So I assumed it should be the correct way to make `post` retry with a fixed retry limit.
```python
retry_function(lambda: self._post_file(file_detail))
```